pkg/ttnpb/field_mask_validation.go

```diff
@@ -0,0 +1,190 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ttnpb
+
+var isEndDeviceReadFieldPaths = []string{
+	"application_server_address",
+	"attributes",
+	"created_at",
+	"description",
+	"ids",
+	"ids.application_ids",
+	"ids.application_ids.application_id",
+	"ids.dev_eui",
+	"ids.device_id",
+	"ids.join_eui",
+	"join_server_address",
+	"locations",
+	"name",
+	"network_server_address",
+	"service_profile_id",
+	"updated_at",
+	"version_ids",
+	"version_ids.brand_id",
+	"version_ids.firmware_version",
+	"version_ids.hardware_version",
+	"version_ids.model_id",
+}
+
+var isEndDeviceWriteFieldPaths = []string{
+	"application_server_address",
+	"attributes",
+	"description",
+	"ids",
+	"ids.dev_eui",
```

Comment:
Doesn't `ids` above imply `ids.*`?

Reply:
No, that simply means that `"ids"` is a valid entry in the `field_mask.paths`

Reply:
Then I will have to add a bunch more fields to NS ones - I was under impression that a specified path would imply all child paths, same as we in utilities in `ttnpb`

Reply:
No, that would result in `ids.some_random_path` being accepted as valid, while it isn't.